### PR TITLE
chore: clean up windows client secret used in building vhd

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
clean up windows client secret used in building vhd.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .pipelines/.vsts-vhd-builder-release-windows.yaml [hunk 1]
```diff
@@ -182,9 +182,6 @@ stages:
             docker run --rm \
             -v ${PWD}:/go/src/github.com/Azure/AgentBaker \
             -w /go/src/github.com/Azure/AgentBaker \
-            -e CLIENT_ID=${AZURE_BUILD_CLIENT_ID} \
```
Comment (cameronmeissner) [inline .pipelines/.vsts-vhd-builder-release-windows.yaml:185]: are you giving packer an MSI now instead? if so how are you accomplishing this?

I've wanted to get this working for our builds for a long time now, but my understanding is that packer doesn't support cross-tenant managed identities

Comment (junjiezhang1997) [reply .pipelines/.vsts-vhd-builder-release-windows.yaml:185]: Yes. I'm giving packer an MSI now. Referring to https://developer.hashicorp.com/packer/integrations/hashicorp/azure#managed-identity, I remove the `client_id`, `client_secret` and `tenant_id`. So packer will attempt to use the Managed Identity and subscription of the VM that Packer is running on, which is related to our 1es hosted pool.

File: vhdbuilder/packer/init-variables.sh [hunk 2]
```diff
@@ -44,15 +44,17 @@ echo "CAPTURED_SIG_VERSION set to: ${CAPTURED_SIG_VERSION}"
...
-if [ -a "${SP_JSON}" ]; then
-	echo "Existing credentials file found."
-	exit 0
-elif [ -z "${CLIENT_ID}" ]; then
-	echo "Service principal not found! Generating one @ ${SP_JSON}"
-	az ad sp create-for-rbac -n aks-images-packer${CREATE_TIME} -o json > ${SP_JSON}
-	CLIENT_ID=$(jq -r .appId ${SP_JSON})
-	CLIENT_SECRET=$(jq -r .password ${SP_JSON})
-	TENANT_ID=$(jq -r .tenant ${SP_JSON})
+if [ "$MODE" == "linuxVhdMode" ]; then
```
Comment (AbelHu) [inline vhdbuilder/packer/init-variables.sh:47]: Can we assure that we have only two modes, linuxVhdMode and windowsVhdMode? If not, I think that it is better to use `"$MODE" != "windowsVhdMode"`

Comment (AbelHu) [reply vhdbuilder/packer/init-variables.sh:47]: @cameronmeissner, it seems like that we only use `linuxVhdMode` and `windowsVhdMode` now and we can remove those legacy code for `default` and `sigMode`. Could you help to confirm it?

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**